docs/js-core-2/week-1/homework.md

```diff
@@ -4,6 +4,190 @@ title: Coursework
+
+// This function is provided. You should not modify it!
+// It takes as input a person -> how it uses the fields
+// in the person object will tell you what fields to put in the
+// objects above
+function logInformationAbout(person) {
+  let age = 2023 - person.yearOfBirth;
+  console.log(
+    `${person.firstName} is from ${person.countryOfOrigin}`,
+    `and will be ${age} years old in 2023`
+  );
+}
+
+logInformationAbout(salem);
+logInformationAbout(ijeoma);
+```
+
+The output should be
+
+```
+Salem is from Sudan and will be 30 years old in 2023
+Ijeoma is from England and will be 22 years old in 2023
+```
+
+1. 1.1 Predict, evaluate and explain the following expressions in the node REPL
+
+```js
+let stock = {
+  name: "aapl",
+  lastClosingPrice: 171.45,
+  lastFiveDaysClosingPrices: [180.33, 176.28, 175.64, 172.99, 171.45],
+};
+
+stock.name;
+stock["name"];
+stock.lastClosingPrice * 100;
+stock.lastFiveDaysClosingPrices[1];
+
+stock.name = "AAPL";
+
+stock;
+```
+
+2. Add a field (a key + value pair) to the stock object, describing it's change over the last 5 days
+
+```js
+let stock = {
+  name: "aapl",
+  lastClosingPrice: 172.99,
+  lastFiveDaysClosingPrices: [179.99, 180.33, 176.28, 175.64, 172.99],
+};
+
+// write code here to add the changeInLastFiveDays field to stock
+
+console.log(
+  "In the last 5 days, the stock changed by",
+  stock.changeInLastFiveDays
+);
+```
+
+The output should be
+
+```
+In the last 5 days, the stock changed by -7
+```
+
+3.1 Write a program that prints out to the console information from the following object
+
+```js
+uk = {
+  name: "The United Kingdom of Great Britain and Northern Ireland",
+  shortName: "The UK",
+  capitalCity: {
+    name: "London",
+    population: 9648000,
+  },
+};
+```
+
+The output should be
+
+```
+The capital of The UK is London. London has a population of 9648000
+```
+
+3.2 Complete the following program so that it prints out information about the stocks that went down in the past 5 days.
+
+```js
+const stocks = [
+  {
+    name: "aapl",
+    lastClosingPrice: 172.99,
+    lastFiveDaysClosingPrices: [179.99, 180.33, 176.28, 175.64, 172.99],
+  },
+  {
+    name: "msft",
+    lastClosingPrice: 327.29,
+    lastFiveDaysClosingPrices: [340.69, 342.45, 334.69, 333.2, 327.29],
+  },
+  {
+    name: "amzn",
+    lastClosingPrice: 3408.34,
+    lastFiveDaysClosingPrices: [3384.44, 3393.39, 3421.37, 3420.74, 3408.34],
+  },
+  {
+    name: "googl",
+    lastClosingPrice: 2869.45,
+    lastFiveDaysClosingPrices: [2951.88, 2958.13, 2938.33, 2928.3, 2869.45],
+  },
+  {
+    name: "tsla",
+    lastClosingPrice: 938.53,
+    lastFiveDaysClosingPrices: [1101.3, 1093.94, 1067.0, 1008.87, 938.53],
+  },
+];
+
+function getMessage(stock) {
+  // return a message containing the name and last closing price of the stock
+}
+
+function isStockPriceDecreasing(stock) {
+  // return a boolean indicating whether the stock decreased in price over the last 5 days
```

Comment:
// return a boolean showing if the stock decreased in price over the last 5 days